pandas/core/reshape/pivot.py

```diff
@@ -79,8 +81,22 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
                 pass
         values = list(values)
 
-    grouped = data.groupby(keys, observed=dropna)
+    # group by the cartesian product of the grouper
+    # if we have a categorical
+    grouped = data.groupby(keys, observed=False)
```

Comment:
I have the feeling this workaround would not be needed if the bug in groupby would be solved? (https://github.com/pandas-dev/pandas/issues/21151)

Reply:
@jreback ?